specification/saml.auth/src/main/java/org/kaazing/specification/saml/internal/Functions.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright 2007-2015, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.specification.saml.internal;
+
+import org.kaazing.k3po.lang.el.Function;
+import org.kaazing.k3po.lang.el.spi.FunctionMapperSpi;
+import org.kaazing.specification.saml.internal.util.AuthnRequestConfiguration;
+import org.kaazing.specification.saml.internal.util.OpenSAMLUtils;
+import org.opensaml.core.config.InitializationException;
+import org.opensaml.saml.saml2.core.AuthnRequest;
+import org.opensaml.saml.saml2.core.Response;
+
+public final class Functions {
+
+    @Function
+    public static String generateEncryptedResponse(String realm, String authnType) throws Exception {
+
+        GenerateResponse.intializeSAML();
+        Response response = GenerateResponse.generateResponse(true);
+
+        String encoded_response = OpenSAMLUtils.compressAndEncodeString(OpenSAMLUtils.SAMLObjectToString(response), false);
+        return String.format("%s realm=\"%s\" saml-authn-request=\"%s\"", authnType, realm, encoded_response);
+    }
+
+    @Function
+    public static String generateResponse(String realm, String authnType) throws Exception {
+
+        GenerateResponse.intializeSAML();
+        Response response = GenerateResponse.generateResponse(false);
+
+        String encoded_response = OpenSAMLUtils.compressAndEncodeString(OpenSAMLUtils.SAMLObjectToString(response), false);
+        return String.format("%s realm=\"%s\" saml-authn-request=\"%s\"", authnType, realm, encoded_response);
+    }
```

Comment:
Because this is how it works...We encode the challenge like this. THe saml-authn-request encoding contains the Base64 encoded Response. This is what we send to the Gateway